authz/middleware.go

```diff
@@ -67,9 +68,16 @@ func (mw *AuthzMiddleware) MiddlewareFunc(h rest.HandlerFunc) rest.HandlerFunc {
 			return
 		}
 
+		// extract resource action
+		action, err := extractResourceAction(r)
```

Comment:
IMO extracResourceAction should be injected into authz middleware, just like resource extraction. Maybe merge that into a single method & struct, for instance

```
type Action stuct {
    Resource string
    Method string
}

type ResourceActionExtractor func(r *rest.Request) (Action, error)
```

Then in middleware.go, populate `Action` fields from X-Original-Method X-Original-URI